deploy/scripts/asc-enable.ps1

```diff
@@ -61,78 +62,79 @@ function Get-AzureRmBearerToken() {
 # Set Azure Sentinel Solution to Log Analytics workspace, enable ASC for IoT and create custom alert for security assessment
 #***************************************************************************************************************************
 function SetSolutionEnableASCForIotAndCreateAlert() {
-	Param(
+    Param(
         [string] $subscriptionId,
-		[string] $resourceGroupName,
-		[string] $workspaceName,
-		[string] $iotHubName
+        [string] $resourceGroupName,
+        [string] $workspaceName,
+        [string] $iotHubName
     )
-	$contentType = "application/json"      
-	$token = Get-AzureRmBearerToken
-	$headers = @{
-		Authorization = $token
-	};
-	$urlPrefix = "https://management.azure.com/subscriptions/" + $subscriptionId + "/resourceGroups/" + $resourceGroupName + "/providers/"; 
-	
-	
-	## add azure sentinel to log analytics workspace
-	$enableSecuritySolutionUrl = $urlPrefix + "Microsoft.OperationsManagement/solutions/SecurityInsights(" + $workspaceName + ")?api-version=2015-11-01-preview";
-	$bodyForSecuritySolution = @{
-	  location = "southeastasia";  
-	  properties = @{
-		  workspaceResourceId = "/subscriptions/" + $subscriptionId + "/resourcegroups/" + $resourceGroupName + "/providers/microsoft.operationalinsights/workspaces/" + $workspaceName;
-		};
-		plan = @{
-		  name = "SecurityInsights(" + $workspaceName + ")";
-		  publisher = "Microsoft";
-		  product = "OMSGallery/SecurityInsights";
-		  promotionCode = ""
-		}
-	};
-	$json = $bodyForSecuritySolution | ConvertTo-Json
-	Invoke-RestMethod -Method PUT -Uri $enableSecuritySolutionUrl -ContentType $contentType -Headers $headers -Body $json;
-	
-	## enable asc for iot and add workspace
-	$iotSecuritysiteName = "AzureIotSolution-" + $workspaceName.substring($workspaceName.length - 5, 5)
-	$enableASCForIotUrl = $urlPrefix + "Microsoft.Security/IoTSecuritySolutions/" + $iotSecuritysiteName + "?api-version=2017-08-01-preview";
-	$bodyForASCForIoT = @{  
-		location = "northeurope";
-		properties = @{       
-			displayName = $iotSecuritysiteName;
-			status = "Enabled";
-			export = @("RawEvents");
-			disabledDataSources = @();
-			workspace = "/subscriptions/" + $subscriptionId + "/resourcegroups/" + $resourceGroupName + "/providers/microsoft.operationalinsights/workspaces/" + $workspaceName;
-			iotHubs = @( "/subscriptions/" + $subscriptionId + "/resourceGroups/" + $resourceGroupName + "/providers/Microsoft.Devices/IotHubs/" + $iotHubName )
-		}
-	};
-	$json2 = $bodyForASCForIoT | ConvertTo-Json
-	Invoke-RestMethod -Method PUT -Uri $enableASCForIotUrl -ContentType $contentType -Headers $headers -Body $json2;
+
+        $resourceGroup = Get-AzureRmResourceGroup -Name $resourceGroupName
```

Comment:
so you only read teh resource group location now from the resourcegroup instead of hard coding it. So all should be fine...

Reply:
yes, correct.